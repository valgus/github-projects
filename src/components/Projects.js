import React, { Component } from "react";
import PropTypes from "prop-types";

import Filter from "./Filter";
import Search from "./Search";
import PTable from "./PTable";

class Projects extends Component {
  constructor(props) {
    super(props);
    const { projects, licenses } = this.props;
    //set all projects to be shown
    const shownProjects = [...projects];
    this.state = {
      licenses,
      shownProjects
    };
    this.onSetSubName = this.onSetSubName.bind(this);
    this.onCheckLicense = this.onCheckLicense.bind(this);
  }

  componentWillReceiveProps(nextProps, nextState) {
    const shownProjects = [...nextProps.projects];
    this.setState({ shownProjects });
  }

  onSetSubName(subname) {
    this.props.onSubnameChange(subname);
  }

  onCheckLicense(i) {
    this.props.onLicenseChange(i);
  }

  render() {
    const { licenses, shownProjects } = this.state;
    const {page, subname} = this.props;
    return (
      <div className="section">
        <div className="columns">
          <div className="column is-four-fifths">
            <Search onChange={this.onSetSubName} value={subname} />
            <div className="is-hidden-tablet is-flex-touch licenses-group">
              <Filter licenses={licenses} onChange={this.onCheckLicense} />
            </div>
            {shownProjects.length === 0 && <p>No projects were found...</p>}
            {shownProjects.length > 0 && <PTable projects={shownProjects} page={page} />}
          </div>
          <div className="column is-one-fifth is-hidden-mobile">
            <Filter licenses={licenses} onChange={this.onCheckLicense} />
          </div>
        </div>
      </div>
    );
  }
}

Projects.propTypes = {
  projects: PropTypes.array.isRequired,
  licenses: PropTypes.array.isRequired,
  subname: PropTypes.string.isRequired,
  page: PropTypes.number.isRequired,
  onLicenseChange: PropTypes.func.isRequired,
  onSubnameChange: PropTypes.func.isRequired
};

export default Projects;
