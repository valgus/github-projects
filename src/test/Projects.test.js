import React from "react";
import Projects from "../components/Projects";
import * as data from "./test-data";
import renderer from "react-test-renderer";

describe("Projects component", () => {
  const projects = data.flatProjects;
  const licenses = data.licenses;
  const subname = "mock_subname";
  const page = 10;
  const onLicenseChange = jest.fn();
  const onSubnameChange = jest.fn();
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Projects
          projects={projects}
          licenses={licenses}
          subname={subname}
          page={page}
          onSubnameChange={onSubnameChange}
          onLicenseChange={onLicenseChange}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
