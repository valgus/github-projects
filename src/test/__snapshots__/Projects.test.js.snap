// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component renders correctly 1`] = `
<div
  className="section"
>
  <div
    className="columns"
  >
    <div
      className="column is-four-fifths"
    >
      <div
        className="field"
      >
        <div
          className="control"
        >
          <input
            className="input is-info is-rounded"
            onChange={[Function]}
            placeholder="Start typing project name..."
            type="text"
            value="mock_subname"
          />
        </div>
      </div>
      <div
        className="is-hidden-tablet is-flex-touch licenses-group"
      >
        <div
          className="license-box"
        >
          <label
            className="checkbox"
          >
            <input
              checked={true}
              id="mit"
              name="mit"
              onChange={[Function]}
              type="checkbox"
            />
            mit
          </label>
        </div>
        <div
          className="license-box"
        >
          <label
            className="checkbox"
          >
            <input
              checked={false}
              id="isc"
              name="isc"
              onChange={[Function]}
              type="checkbox"
            />
            isc
          </label>
        </div>
        <div
          className="license-box"
        >
          <label
            className="checkbox"
          >
            <input
              checked={true}
              id="apache-2.0"
              name="apache-2.0"
              onChange={[Function]}
              type="checkbox"
            />
            apache-2.0
          </label>
        </div>
      </div>
      <table
        className="table is-fullwidth is-narrow"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              stars
            </th>
            <th
              className="is-hidden-touch"
            >
              owner
            </th>
            <th>
              name
            </th>
            <th
              className="is-hidden-touch"
            >
              description
            </th>
            <th
              className="is-hidden-mobile"
            >
              license
            </th>
            <th
              className="is-hidden-touch"
            >
              created at
            </th>
            <th>
              link
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              91
            </td>
            <td>
              2182
            </td>
            <td
              className="is-hidden-touch"
            >
              danburzo
            </td>
            <td
              className="project-name"
            >
              percollate
            </td>
            <td
              className="is-hidden-touch"
            >
              üåê ‚Üí üìñ A command-line tool to turn web pages into beautifully formatted PDFs
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-09-28
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/danburzo/percollate"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              92
            </td>
            <td>
              917
            </td>
            <td
              className="is-hidden-touch"
            >
              rhardih
            </td>
            <td
              className="project-name"
            >
              ekill
            </td>
            <td
              className="is-hidden-touch"
            >
              Chrome extension to nuke annoying elements in a web page
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-10-02
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/rhardih/ekill"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              93
            </td>
            <td>
              722
            </td>
            <td
              className="is-hidden-touch"
            >
              oscarmorrison
            </td>
            <td
              className="project-name"
            >
              md-page
            </td>
            <td
              className="is-hidden-touch"
            >
              üìù create a webpage with just markdown
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-10-05
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/oscarmorrison/md-page"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              94
            </td>
            <td>
              476
            </td>
            <td
              className="is-hidden-touch"
            >
              nodkz
            </td>
            <td
              className="project-name"
            >
              lvovich
            </td>
            <td
              className="is-hidden-touch"
            >
              –°–∫–ª–æ–Ω–µ–Ω–∏–µ –Ω–∞–∑–≤–∞–Ω–∏–π –≥–æ—Ä–æ–¥–æ–≤, –æ–ø—Ä–µ–¥–µ–ª–µ–Ω–∏—è –ø–æ–ª–∞ –ø–æ –§–ò–û, —Å–∫–ª–æ–Ω–µ–Ω–∏—è –∏–º–µ–Ω –ø–æ –ø–∞–¥–µ–∂–∞–º
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-09-25
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/nodkz/lvovich"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              95
            </td>
            <td>
              313
            </td>
            <td
              className="is-hidden-touch"
            >
              callstack
            </td>
            <td
              className="project-name"
            >
              react-native-testing-library
            </td>
            <td
              className="is-hidden-touch"
            >
              Simple React Native testing utilities helping you write better tests with less effort
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-10-01
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/callstack/react-native-testing-library"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              96
            </td>
            <td>
              252
            </td>
            <td
              className="is-hidden-touch"
            >
              victorqribeiro
            </td>
            <td
              className="project-name"
            >
              imgToAscii
            </td>
            <td
              className="is-hidden-touch"
            >
              A JavaScript implementation of a image to Ascii code
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-10-03
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/victorqribeiro/imgToAscii"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              97
            </td>
            <td>
              189
            </td>
            <td
              className="is-hidden-touch"
            >
              burakcan
            </td>
            <td
              className="project-name"
            >
              mb
            </td>
            <td
              className="is-hidden-touch"
            >
              Exception-free nested nullable attribute accessor
            </td>
            <td
              className="is-hidden-mobile"
            >
              The Unlicense
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-10-10
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/burakcan/mb"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              98
            </td>
            <td>
              172
            </td>
            <td
              className="is-hidden-touch"
            >
              kingzez
            </td>
            <td
              className="project-name"
            >
              write-readable-javascript-code
            </td>
            <td
              className="is-hidden-touch"
            >
              üìñ All about writing maintainable JavaScript
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-09-27
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/kingzez/write-readable-javascript-code"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              99
            </td>
            <td>
              162
            </td>
            <td
              className="is-hidden-touch"
            >
              sindresorhus
            </td>
            <td
              className="project-name"
            >
              filter-console
            </td>
            <td
              className="is-hidden-touch"
            >
              Filter out unwanted \`console.log()\` output
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-10-10
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/sindresorhus/filter-console"
              >
                Link
              </a>
            </td>
          </tr>
          <tr>
            <td>
              100
            </td>
            <td>
              147
            </td>
            <td
              className="is-hidden-touch"
            >
              EyalPerry
            </td>
            <td
              className="project-name"
            >
              dev-cv
            </td>
            <td
              className="is-hidden-touch"
            >
              A CV Template for Software Developers
            </td>
            <td
              className="is-hidden-mobile"
            >
              MIT License
            </td>
            <td
              className="is-hidden-touch"
            >
              2018-10-06
            </td>
            <td>
              <a
                className="button is-info"
                href="https://github.com/EyalPerry/dev-cv"
              >
                Link
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="column is-one-fifth is-hidden-mobile"
    >
      <div
        className="license-box"
      >
        <label
          className="checkbox"
        >
          <input
            checked={true}
            id="mit"
            name="mit"
            onChange={[Function]}
            type="checkbox"
          />
          mit
        </label>
      </div>
      <div
        className="license-box"
      >
        <label
          className="checkbox"
        >
          <input
            checked={false}
            id="isc"
            name="isc"
            onChange={[Function]}
            type="checkbox"
          />
          isc
        </label>
      </div>
      <div
        className="license-box"
      >
        <label
          className="checkbox"
        >
          <input
            checked={true}
            id="apache-2.0"
            name="apache-2.0"
            onChange={[Function]}
            type="checkbox"
          />
          apache-2.0
        </label>
      </div>
    </div>
  </div>
</div>
`;
